v2/internal/testcommon/vcr/redact.go

```diff
@@ -44,21 +63,50 @@ var responseHeadersToRemove = []string{
 	"Date",
 }
 
-func RedactResponseHeaders(headers http.Header) {
+func RedactResponseHeaders(azureIDs creds.AzureIDs, headers http.Header) {
+	// Hide sensitive response headers
+	for key, values := range headers {
+		for i := range values {
+			values[i] = strings.ReplaceAll(values[i], azureIDs.SubscriptionID, nilGuid)
+			values[i] = strings.ReplaceAll(values[i], azureIDs.TenantID, nilGuid)
+			if azureIDs.BillingInvoiceID != "" {
+				values[i] = strings.ReplaceAll(values[i], azureIDs.BillingInvoiceID, creds.DummyBillingId)
+			}
+		}
+
+		// Hide the base request URL in the AzureOperation and Location headers
+		if key == genericarmclient.AsyncOperationHeader || key == genericarmclient.LocationHeader {
+			for i := range values {
+				values[i] = HideURLData(azureIDs, values[i])
+			}
+		}
+	}
+
 	for _, header := range responseHeadersToRemove {
 		delete(headers, header)
 	}
 }
 
-func HideRecordingData(s string) string {
-	result := hideDates(s)
-	result = hideSSHKeys(result)
-	result = hidePasswords(result)
-	result = hideKubeConfigs(result)
-	result = hideKeys(result)
-	result = hideCustomKeys(result)
+func HideRecordingData(azureIDs creds.AzureIDs, s string) string {
+	// Hide the subscription ID
+	s = strings.ReplaceAll(s, azureIDs.SubscriptionID, nilGuid)
```

Comment:
heh, I actually removed `result` because it caused me bugs.

The bug was:
```
result := hideThing1(s)

result = hideThing2(s) // oops, just dropped the result of hideThing1
```

It seems easier to me, at least in this context, if there's only a single string we're operating on to avoid mistakes like the above. I definitely agree that there are also cases where the reverse is true, but I think usually those cases involve caring about both the pre-modified variable and the post-modified variable in a way that we don't here.

Here we're basically wanting to write `replaceThing1(replaceThing2(replaceThing3(replaceThing4(input))))` but in a less ugly way. If we were in a more functional language we might write something like: `replaceThing1 . replaceThing2 . replaceThing3` (where `.` was function composition).